### PR TITLE
Merge dev

### PR DESCRIPTION
**Pull request by jmg-duarte**

### CODE REVIEW

File: rust/sdk/Cargo.toml [hunk 1]
```diff
@@ -12,79 +12,78 @@ license = "Apache-2.0"
...
+# This one is to make examples work
+asynchronous-codec = "0.7.0"
+structopt = "0.3.26"
+chrono = "0.4.31"
 
 [[example]]
 name = "files"
 path = "examples/files.rs"
-required-features = ["client"]
+required-features = ["with_tokio"]
```
Comment (Kelerchian) [inline rust/sdk/Cargo.toml:54]: leftover "with_tokio" feature flag

File: rust/actyx/ax/tests/smoke.rs
Comment (jmg-duarte) [inline rust/actyx/ax/tests/smoke.rs:1]: Why the change from `localhost`?

File: rust/actyx/ax-types/src/timestamp.rs [hunk 2]
```diff
@@ -54,7 +55,9 @@ impl TryFrom<SystemTime> for Timestamp {
 
 impl From<Timestamp> for DateTime<Utc> {
     fn from(ts: Timestamp) -> DateTime<Utc> {
-        Utc.timestamp((ts.0 / 1_000_000) as i64, (ts.0 % 1_000_000) as u32 * 1000)
+        TimeZone::timestamp_micros(&Utc, ts.0 as i64)
+            .single()
+            .expect("Timestamp out of range")
```
Comment (rkuhn) [inline rust/actyx/ax-types/src/timestamp.rs:60]: as this is a general facility and the user may supply any u64 timestamp, this should probably be turned into TryFrom; I’m looking into this now

File: rust/actyx/ax-core/src/libp2p_streaming_response/protocol_v2.rs [hunk 3]
```diff
@@ -80,30 +79,32 @@ impl ProtocolError {
...
-        write_err(io, &err).await?;
-        return Err(err);
-    }
-    let size = buffer.len() - 4;
-    if size > (max_size as usize) {
-        tracing::debug!("message size {} too large (max = {})", size, max_size);
-        let err = ProtocolError::MessageTooLargeSent(size);
-        write_err(io, &err).await?;
-        return Err(err);
+    async move {
```
[the remark this answers is not in the document]
Comment (rkuhn) [reply rust/actyx/ax-core/src/libp2p_streaming_response/protocol_v2.rs:90]: because the previous formulation was `!Send`

File: rust/actyx/ax-core/src/util/serde_support.rs
[the remark this answers is not in the document]
Comment (rkuhn) [reply rust/actyx/ax-core/src/util/serde_support.rs:1]: :+1:

**Review by rkuhn — COMMENTED** [review]
What is left now is mostly to remove the unused utilities. I opened an issue to create READMEs afterwards.

File: rust/actyx/ax-core/src/util/chunk_unless_pending.rs
[the remark this answers is not in the document]
Comment (jmg-duarte) [reply rust/actyx/ax-core/src/util/chunk_unless_pending.rs:1]: Seems it wasn't even being used

File: rust/actyx/ax-core/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+[AX](https://crates.io/crates/ax) is a decentralized event database, streaming and processing
+engine that allows you to easily build [local-first cooperative](https://www.local-first-cooperation.org/)
+apps.
+It makes it easy to run distributed applications on multiple nodes.
+It is a piece of software that allows you to run your own apps on one or more edge devices and have these apps seamlessly communicate and share data with each other.
+
+---
+
+> **Note:** _AX has been created by the company [Actyx AG](https://developer.actyx.com/) and predates the unrelated [Actix framework](https://crates.io/crates/actix) (although AX wasn’t open-sourced until Oct’23).
+> While we have changed crate names to avoid confusion, the company name remains Actyx AG and appears in some places._
```
Comment (jmg-duarte) [inline rust/actyx/ax-core/README.md:15]: A quick "if you find "Actyx" where "AX" should be, let us know" is a small call to action that _may_ (unlikely but still) lead people to engage

File: rust/actyx/ax/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+> **Note:** _AX has been created by the company [Actyx AG](https://developer.actyx.com/) and predates the unrelated [Actix framework](https://crates.io/crates/actix) (although AX wasn’t open-sourced until Oct’23).
+> While we have changed crate names to avoid confusion, the company name remains Actyx AG and appears in some places._
+
+---
+
+## Installation
+
+This is a binary crate, its intended use is via
+
+    cargo install ax
```
Comment (jmg-duarte) [inline rust/actyx/ax/README.md:23]: Not a fan of this style of code blocks, especially given that most (if not all) code blocks we have use triple backticks

File: rust/sdk/src/client.rs [hunk 6]
```diff
@@ -1089,8 +1089,8 @@ impl<'a> FusedFuture for Subscribe<'a> {
 /// Request builder for monotonic subscriptions.
 ///
 /// Monotonic subscriptions keep track of the highest sort order
-/// ([`LamportTimestamp`](crate::timestamp::LamportTimestamp) and
-/// [`StreamId`](crate::scalars::StreamId)) seen so far, ending the stream with a
+/// ([`LamportTimestamp`](crate::types::LamportTimestamp) and
+/// [`StreamId`](crate::types::StreamId)) seen so far, ending the stream with a
```
Comment (jmg-duarte) [inline rust/sdk/src/client.rs:1093]: Note: I'm still not sure about these names or import paths, should we discuss or do you guys like them?

**Review by jmg-duarte — COMMENTED** [review]
Looking good!
